agent/ui_endpoint.go

```diff
@@ -343,6 +343,20 @@ RPC:
 		}
 		upstreamResp = append(upstreamResp, &sum)
 	}
+	for k, v := range out.ServiceTopology.UpstreamSources {
+		if v == structs.TopologySourceRoutingConfig {
+			sn := structs.ServiceNameFromString(k)
+			sum := ServiceTopologySummary{
+				ServiceSummary: ServiceSummary{
```

Comment:
I'm not a fan of pretending this is a service but I've created a dummy ServiceSummary for the response body, since intention and source get flattened into a ServiceTopologySummary object per "service". This is least friction/effort for frontend as well.